### PR TITLE
Adding tests

### PR DESCRIPTION
**Pull request by cxoltero**
There is an issue when trying to unit test for errors inside of customReporter method. 
logs are not added to the log array for spies.

### CODE REVIEW

**Comment by obuckley-kenzan** [comment]
Can one of the admins verify this patch?

**Comment by obuckley-kenzan** [comment]
@cxoltero 
Please make sure you are referencing issues in the queue when creating PRs, ideally in the branch name, eg. something like `issue/4-add-tests`

I've been meaning to add a CONTRIBUTING.md so apologies for not getting it in there sooner.

**Comment by obuckley-kenzan** [comment]
resolves #3

File: gulpfile.js [hunk 1]
```diff
@@ -2,11 +2,25 @@
 
 var gulp = require('gulp');
 var validateCssPipeline = require('./src/index.js')();
-var validatePipeline = require('pipeline-validate-js')();
+var validatePipeline = require('pipeline-validate-js');
+var testPipeline = require('pipeline-test-node')({ plugins: {
+  mocha: {
```
Comment (obuckley-kenzan) [inline gulpfile.js]: @paul-barry-kenzan 
Seems like we should get some of these as default options in the the pipeline, yes?

File: gulpfile.js [hunk 2]
```diff
@@ -2,11 +2,25 @@
...
+    reporter: 'spec'
+  },
+  istanbul: {
+    reporters: ['text-summary'],
+    thresholds: {
+      global: 60
+    }
+  }
+}});
+var fs = require('fs');
```
Comment (obuckley-kenzan) [inline gulpfile.js]: I prefer to alpha sort variables

**Comment by obuckley-kenzan** [comment]
@cxoltero 
Thanks for this contribution @cxoltero , just a couple feedback items but overall this is looking good to me!

**Comment by cxoltero** [comment]
@obuckley-kenzan, as of this last PR, it seems like issue#7 also gets fixed. I do not get the build task to finish executing. it errors out. 
screenshot 2016-03-30 09 57 02

**Comment by obuckley-kenzan** [comment]
awesome!

**Comment by obuckley-kenzan** [comment]
ok to test

**Pull request merged**